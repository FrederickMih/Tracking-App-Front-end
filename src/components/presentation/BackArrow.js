import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import '../../styles/BackArrow.css';
import ArrowBackIosOutlinedIcon from '@material-ui/icons/ArrowBackIosOutlined';

const AppBar = ({ link }) => (
  <>
    <div className="bkbar">
      <Link to={link}>
        <ArrowBackIosOutlinedIcon />
      </Link>
    </div>
  </>
);

AppBar.propTypes = {
  link: PropTypes.string,
};

AppBar.defaultProps = {
  link: '/add',
};

export default AppBar;
