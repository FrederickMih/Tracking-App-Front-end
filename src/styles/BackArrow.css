.bkbar {
  background-color: var(--accent-primary);
  display: flex;
  padding: 1.2em 1em 0.6em;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
}

.bkbar a {
  color: #eee;
  position: absolute;
  left: 0.8em;
}
