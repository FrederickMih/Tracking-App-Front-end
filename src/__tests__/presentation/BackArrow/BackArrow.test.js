import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../Redux/configureStore';
import BackArrow from '../../../components/presentation/BackArrow';

const store = configureAppStore();

describe('BackArrow', () => {
  test('should match the snapshot of BackArrow', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <BackArrow title="bkbar" />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
