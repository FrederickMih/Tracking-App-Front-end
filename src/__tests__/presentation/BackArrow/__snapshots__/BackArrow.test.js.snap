// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackArrow should match the snapshot of BackArrow 1`] = `
<div
  className="bkbar"
>
  <a
    href="/add"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
      />
    </svg>
  </a>
</div>
`;
